### PR TITLE
[ci] Upload build asset data to darc/maestro

### PR DESCRIPTION
**Pull request by pjcollins**
The `maui-release` pipeline has been updated to publish information
about MAUI .NET .nupkg files to the dotnet Maestro server.

With this data being published, the `darc` tool can now be used to
manage a MAUI .NET  product dependency.

For dependency updating to work, we must configure "default channel"
associations for all MAUI branches that other products may be interested
in tracking.  The `darc` tool can be used to do this manually:

    darc add-default-channel --channel ".NET 9.0.1xx SDK" --branch "net9.0" --repo https://github.com/dotnet/maui

This default channel configuration ensures that all new builds produced
by the net9.0 branch will be associated with the .NET 9 channel.  MAUI
consumers can then update to the latest build available from net9.0 with
the `update-dependencies` command:

    darc update-dependencies --channel ".NET 9.0.1xx SDK"

### CODE REVIEW

File: eng/pipelines/common/sdk-insertion.yml [hunk 1]
```diff
@@ -57,3 +61,21 @@ jobs:
...
+        -p:BuildAssetRegistryToken=$(MaestroAccessToken)
+        -p:OutputPath=$(Build.StagingDirectory)\nuget\
+        -v:n -bl:$(Build.StagingDirectory)\binlogs\push-bar-manifest.binlog
+    condition: and(succeeded(), eq('${{ parameters.pushMauiPackagesToMaestro }}', 'true'))
+  - powershell: |
+      $versionEndpoint = 'https://maestro-prod.westus2.cloudapp.azure.com/api/assets/darc-version?api-version=2019-01-16'
+      $darcVersion = $(Invoke-WebRequest -Uri $versionEndpoint -UseBasicParsing).Content
+      $arcadeServicesSource = 'https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json'
+      & dotnet tool update microsoft.dotnet.darc --version "$darcVersion" --add-source "$arcadeServicesSource" --tool-path $(Agent.ToolsDirectory)\darc -v n
+      & $(Agent.ToolsDirectory)\darc\darc add-build-to-channel --default-channels --id $(BARBuildId) --publishing-infra-version 3 --skip-assets-publishing --password $(MaestroAccessToken) --azdev-pat $(publishing-dnceng-devdiv-code-r-build-re)
```
Comment (pjcollins) [inline eng/pipelines/common/sdk-insertion.yml:79]: With these changes in place, we will need to make sure to manually create "default channels" for MAUI for every major branch, and also every time we create a new release/servicing branch:

    darc add-default-channel --channel ".NET 9.0.1xx SDK" --branch "net9.0" --repo https://github.com/dotnet/maui

**Comment by pjcollins** [comment]
Looks like this test build worked: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9125468&view=logs&j=d60cae5e-2024-5790-45cc-e679bfe2722d

And I just pushed the dummy build to a test channel (since we have no default channels configured) - https://maestro-prod.westus2.cloudapp.azure.com/1647/https:%2F%2Fgithub.com%2Fdotnet%2Fmaui/latest/graph

File: eng/pipelines/common/sdk-insertion.yml (hunk 1 above)
Comment (jonathanpeppers) [reply eng/pipelines/common/sdk-insertion.yml:78]: I wrote down this command in a doc like this:

* https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/HowToBranch.md

Comment (rmarinho) [reply eng/pipelines/common/sdk-insertion.yml:78]: Yeah right now this is fine.. We can also add to the loop. @PureWeen  knows the correct place for sure. He as looking at organising better our release process. Should it be on the wiki to be public?

**Review by rmarinho — APPROVED** [review]

**Pull request merged**